Assets/HoloToolkit/SpatialMapping/Scripts/SpatialMappingObserver.cs

```diff
@@ -107,22 +85,72 @@ protected override void Awake()
         /// </summary>
         private void Update()
         {
-            // Only do processing if the observer is running.
-            if (ObserverState == ObserverStates.Running)
+            if ((ObserverState == ObserverStates.Running) && (outstandingMeshRequest == null))
             {
-                // If we don't have mesh creation in flight, but we could schedule mesh creation, do so.
-                if (surfaceWorkOutstanding == false && surfaceWorkQueue.Count > 0)
+                if (surfaceWorkQueue.Count > 0)
                 {
-                    // Pop the SurfaceData off the queue.  A more sophisticated algorithm could prioritize
+                    // We're using a simple first-in-first-out rule for requesting meshes, but a more sophisticated algorithm could prioritize
                     // the queue based on distance to the user or some other metric.
-                    SurfaceData surfaceData = surfaceWorkQueue.Dequeue();
+                    var surfaceID = surfaceWorkQueue.Dequeue();
+
+                    string surfaceName = ("Surface-" + surfaceID.handle);
+
+                    SurfaceObject newSurface;
+                    WorldAnchor worldAnchor;
+                    
+                    if (spareSurfaceObject == null)
+                    {
+                        newSurface = CreateSurfaceObject(
+                            mesh: null,
+                            objectName: surfaceName,
+                            parentObject: transform,
+                            meshID: surfaceID.handle,
+                            drawVisualMeshesOverride: false
+                            );
+
+                        worldAnchor = newSurface.Object.AddComponent<WorldAnchor>();
+                    }
+                    else
+                    {
+                        newSurface = spareSurfaceObject.Value;
+                        spareSurfaceObject = null;
+
+                        Debug.Assert(!newSurface.Object.activeSelf);
+                        newSurface.Object.SetActive(true);
 
-                    // If RequestMeshAsync succeeds, then we have successfully scheduled mesh creation.
-                    surfaceWorkOutstanding = observer.RequestMeshAsync(surfaceData, SurfaceObserver_OnDataReady);
+                        Debug.Assert(newSurface.Filter.sharedMesh == null);
+                        Debug.Assert(newSurface.Collider.sharedMesh == null);
+                        newSurface.Object.name = surfaceName;
+                        Debug.Assert(newSurface.Object.transform.parent == transform);
+                        newSurface.ID = surfaceID.handle;
+                        newSurface.Renderer.enabled = false;
+
+                        worldAnchor = newSurface.Object.GetComponent<WorldAnchor>();
+                        Debug.Assert(worldAnchor != null);
+                    }
+
+                    var surfaceData = new SurfaceData(
```

Comment:
This one I'd like to keep as "var".  Especially when the type is elsewhere on the declaring line, I like to avoid specifying it twice.  How do others feel?